README.md

```diff
@@ -1,2 +1,106 @@
-# guidebook_challange
-A simple application to create a DB schema and load data from csv files (Guidebook coding challenge)
+# gbapp
+
+A simple application to create a DB table and load data from csv files
+(GB coding challenge)
+
+## Getting started
+
+Clone this project locally
+
+```
+
+git clone git@github.com:vishket/gbapp.git
+
+```
+
+### Pre requisites
+
+1) This application requires **Python 2.7 +**, stable version of **Flask**
+, the Python library for postgres **psycopg2** and the dotenv module
+**python-dotenv**
+
+These dependencies can be easily installed by running the requirement.txt
+ file like so...
+
+```
+mkvirtualenv vishket
+
+pip install -r requirements.txt
+```
+
+2) Database connection parameters
+
+Enter your database connection values in the .env file which resides at
+the root of the application folder
+
+## Run
+
+Once you have all the dependencies installed, simply...
+
+```
+python gbapp/app.py
+```
+
+This will launch the application and start serving locally on the
+default port 5000
+
+To view the app go to http://localhost:5000/
+
+## Features
+
+This application provides two endpoints
+
+- Create a Database table: http://localhost:5000/create-table
+
+The *create-table* endpoint creates a new table based on the schema
+defined in the 'schema.csv' file. You can also navigate to the same
+endpoint from the hyperlink on the home page.
+
+WARNING: Creating a table with the same name as an existing one would
+result in an error. You will either have to DROP the existing table OR
+rename the new table in the .env file
+
+- Load data into table: http://localhost:5000/load-data
+
+The *load-data* endpoint loads data from 'data.csv' into an existing
+database table.
+
+## Examples
+
+The drop_data folder consists of a couple of sample schema,data csv
+pairs. By default, the app will look for files named "schema.csv" and
+"data.csv". In order to run the second pair of files, you will have
+to swap the filenames
+
+
+## Tests
+
+Unit tests can be found in the test folder. Tests can be executed by
+simply running...
+
+```
+pytest
+```
+
+## Questions & Assumptions
+
+1) Column width:
+
+The "width" field in the example schema.csv file seemed a bit
```

Comment:
Great feedback -- I appreciate you being communicative about them in your documentation 👍